crates/ruff_linter/src/rules/pylint/rules/boolean_chained_comparison.rs

```diff
@@ -83,20 +87,30 @@ pub(crate) fn boolean_chained_comparison(checker: &mut Checker, expr_bool_op: &E
                 return None;
             }
 
-            let edit = Edit::range_replacement(
-                left_compare_right.id().to_string(),
-                TextRange::new(left_compare_right.start(), right_compare_left.end()),
+            let replace_range =
+                TextRange::new(left_compare_right.start(), right_compare_left.end());
+
+            // It's unsafe to apply the fix if the range contains a parenthesis
+            let fix = if locator.slice(replace_range).contains(')') {
```

Comment:
This is probably fine considering that we're opting out of a fix but you can use `SimpleTokenizer` to get proper lexing. But the solution might actually be to use `parenthesized_range` which gives you the parenthesized expression range.

Reply:
Ah `parenthesized_range` is my dream

Reply:
This still looks hard to get quite right. What do you think about patching the bug then dealing with adding a fix later?

Reply:
I would need to look into what's needed to fix this but I would prefer using `parentheses_iterator` to detect if the left or right are parenthesized if that's possible.

Reply:
Sure, we can do that at least.

Reply:
Added that, and we'll offer a fix if they're both parenthesized.